src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,50 @@
 'use strict';
 
+const HEALTH = 100;
+const DAMAGE_PER_BITE = 50;
+
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = HEALTH;
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+    Animal.alive.push(this);
+  }
+
+  hide() {
+    this.hidden = !this.hidden;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    Animal.alive.push(this);
+  }
+
+  bite(target) {
+    if (
+      target.health > 0
+      && target instanceof Herbivore
+      && !target.hidden
+    ) {
+      target.health -= DAMAGE_PER_BITE;
+
+      if (target.health === 0 || target.health < 0) {
```

Comment:
```suggestion
      if (target.health <= 0) {
```
It can be simplified